website/s3_logger.py

```diff
@@ -1,38 +1,70 @@
 import logging
+import json
+import os
 
 from . import aws_helpers, log_queue
 
 logger = logging.getLogger(__name__)
 
 
+should_log_activity = os.getenv("LOG_USER_ACTIVITY")
+
+
 class NullLogger:
     """A logger that doesn't actually do anything."""
 
+    def __init__(self, file_path="activity_data.json", **kwargs):
+        self.is_tracking = kwargs.get("tracking")
+        self.file_path = file_path
```

Comment:
`NullLogger` is intended to mean "have the classes, but don't actually log".

If you change the `NullLogger` to log to a file, it's not actually a `NullLogger` anymore. Wouldn't you want to add a `FileLogger` instead?

Then, depending on `should_log_activity` (or potentially `utils.is_heroku()`) you can decide to construct one of `NullLogger`, `FileLogger` or `S3Logger` at startup. 

Here is how this was intended to work:

![image](https://github.com/hedyorg/hedy/assets/524162/361f441f-2365-4e1f-91a0-a21b6ec9ae68)

The idea is that we have an interface (which of course doesn't existing in Python because it's all duck-typed), which is just an object with a `log()` method, and then we'd construct one of 3 objects that satisfy that interface: a `NullLogger` which doesn't do anything, a `FileLogger` which writes to a file, or an `S3Logger` which writes to S3 by means of a log queue that sits in the background and periodically writes to S3 (because again we want to batch up little writes when we do network calls).

Reply:
Of course, we could also have done the following:

![image](https://github.com/hedyorg/hedy/assets/524162/6a2efe8f-e58e-478f-bc6c-fa5b27547ac8)

There is a `Logger` object that is concrete and that every caller interacts with, but the transmission strategy varies (either do nothing, write to a file, or send to S3).

The designs don't differ that much, the only difference is that in the 2nd we have a place to put some common logic between the 3 loggers, if we need to. I don't know that we need this, so that's why so far in my head we would just have different Loggers themselves.

Reply:
I made some changes @rix0rrr! 
- created a FileLogger on top the NullLogger.
- moved code logic appropriately.